### DISCUSSION TITLE
Problem with commands in PHP

### DISCUSSION THREAD

**Discussion by lortega14** [cyanfish/naps2#448] [Development]
Hello, I wanted to implement NAPS2 console on Ubuntu with PHP by executing a command from the PHP code, but for some reason, I get a response saying that I am not selecting the profile or that it is ambiguous, and it doesn't detect the profiles when I run the command in PHP. However, when I execute it in the terminal, it scans without any issues. Any help?

Best regards.

**Comment by cyanfish** [comment]
Profiles are stored at the user level, I assume your PHP is running as a different user. You can use the [--noprofile](https://www.naps2.com/doc/command-line#profile-options) option to [specify](https://www.naps2.com/doc/command-line#scanning-options) settings on the command line instead of using a profile.

**Reply by lortega14** [reply]
I tried it, but when I attempted to do it, the message "No device was specified. Either use "--profile" to specify a profile with a device, or use "--device" to choose a particular device (use "--listdevices" to see available choices)" appeared. And when I used 'profile,' the initial problem came back. Any suggestions?

**Reply by cyanfish** [reply]
As the message says, you need to use "--device" to specify which device to scan with.

**Reply by lortega14** [reply]
i tried whit this "$command='/usr/lib/naps2/naps2 console --driver sane --device "S2000 (kds_s2000)" --source duplex -o "/var/www/html/naps2/escaneados/prueba.pdf" --force';" and the initial problem came back. ):

**Reply by cyanfish** [reply]
You're missing --noprofile.
